target_chains/aptos/contracts/sources/merkle.move

```diff
@@ -0,0 +1,59 @@
+module pyth::merkle {
+    use std::vector;
+    use pyth::keccak160;
+    use pyth::keccak160::Hash;
+
+    const LEAF_PREFIX: u8 = 0;
+    const NODE_PREFIX: u8 = 1;
+
+
+    fun hash_leaf(leaf: vector<u8>): Hash {
+        let leaf_data: vector<u8> = vector[LEAF_PREFIX];
+        vector::append(&mut leaf_data, leaf);
+        keccak160::from_data(leaf_data)
+    }
+
+    fun hash_node(node1: Hash, node2: Hash): Hash {
+        let node_data: vector<u8> = vector[NODE_PREFIX];
+
+        if (keccak160::is_smaller(&node2, &node1)) {
+            vector::append(&mut node_data, keccak160::get_data(&node2));
+            vector::append(&mut node_data, keccak160::get_data(&node1));
+        }
+        else {
+            vector::append(&mut node_data, keccak160::get_data(&node1));
+            vector::append(&mut node_data, keccak160::get_data(&node2));
+        };
+        keccak160::from_data(node_data)
+    }
+
+    public fun check(path: &vector<vector<u8>>, root: &vector<u8>, _leaf: vector<u8>): bool {
+        let i = 0;
+
+        let current = hash_leaf(_leaf);
+        while (i < vector::length(path)) {
+            let node = keccak160::new(*vector::borrow(path, i));
+            current = hash_node(current, node);
+            i = i + 1;
+        };
+        &current == &keccak160::new(*root)
```

Comment:
should one check that the references are equal, or that the underlying bytes are equal?